### PR TITLE
Migrate pipelines from TravisCI to Github Actions

### PR DESCRIPTION
**Pull request by hugomiguelabreu**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: Build and test
+        run: mvn test -B
+
+      - uses: codecov/codecov-action@v1
+        with:
+          fail_ci_if_error: true
```
Comment (JPDSousa) [inline .github/workflows/build.yml:44]: Add empty line

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Push openml-java packages to maven central repository
+
+on:
+  push:
+    tags:
+      - '*.*.*'
```
Comment (JPDSousa) [inline .github/workflows/deploy.yml:6]: We should also allow for experimental versions, either based on the classic `xp-*` branch glob, or the `*.*.*-*` tag (which is actually the one we're using above, so we might just go with that one)

**Review by JPDSousa — COMMENTED** [review]
Did you test triggering an experimental release? I think that would help us figure out if the deploy job is also correctly configured.

**Comment by hugomiguelabreu** [comment]
Added a experimental tag to test the release action.

**Review by JPDSousa — APPROVED** [review]

**Pull request merged**